doozerlib/cli/release_gen_payload.py

```diff
@@ -1279,6 +1292,44 @@ def find_rhcos_build_rpm_inconsistencies(rhcos_builds: List[RHCOSBuildInspector]
         # Report back rpm name keys which were associated with more than one NVR in the set of RHCOS builds.
         return {rpm_name: nvr_dict for rpm_name, nvr_dict in rpm_uses.items() if len(nvr_dict) > 1}
 
+    @staticmethod
+    def find_rhcos_build_kernel_inconsistencies(rhcos_build: RHCOSBuildInspector) -> List[Dict[str, str]]:
+        """
+        Looks through a set of RHCOS builds and finds if any of those builds contains a kernel-rt version that
+        is inconsistent with kernel.
+
+        e.g. kernel-4.18.0-372.43.1.el8_6 and kernel-rt-4.18.0-372.43.1.rt7.200.el8_6 are consistent,
+        while kernel-4.18.0-372.43.1.el8_6 and kernel-rt-4.18.0-372.41.1.rt7.198.el8_6 are not.
+        :return: Returns List[Dict[inconsistent_rpm_name, rpm_nvra]] The List will be empty
+                 if there are no inconsistencies detected.
+        """
+        inconsistencies = []
+        rpm_list = rhcos_build.get_os_metadata_rpm_list()
+        rpms_dict = {entry[0]: entry for entry in rpm_list}
+
+        def _to_nvr(nevra):
+            return f'{nevra[0]}-{nevra[2]}-{nevra[3]}'
+
+        if {"kernel-core", "kernel-rt-core"} <= rpms_dict.keys():
+            kernel_v, kernel_r = rpms_dict["kernel-core"][2:4]
```

Comment:
Hard to understand what we are indexing, can you provide an example? Like you did at line 1320